common/datavalues/src/arrays/boolean/mutable.rs

```diff
@@ -71,15 +78,27 @@ impl MutableBooleanArrayBuilder {
         Self {
             data_type: DataType::Boolean,
             values: MutableBitmap::with_capacity(capacity),
-            validity: None,
+            validity: Some(MutableBitmap::with_capacity(capacity)),
         }
     }
 
-    pub fn from_data(values: MutableBitmap, validity: Option<MutableBitmap>) -> Self {
+    pub fn push(&mut self, value: bool) {
+        self.values.push(value);
+        self.validity.as_mut().unwrap().push(true);
```

Comment:
Maybe no need to care about `validity` for `not nullable` case, the validity is always none.

Reply:
yes, it it redundant.